rust/src/pool/node_pool_with_storage.rs

```diff
@@ -0,0 +1,84 @@
+use std::ops::{Deref, DerefMut};
+
+use crate::types::Node;
+
+/// A [`Node`] with a **status** to store metadata about the node lifecycle.
+/// The internal node is set as [`NodeStatus::Dirty`] when a mutable reference is requested.
+/// Accessing a deleted node will panic.
```

Comment:
```suggestion
/// A [`Node`] with a **status** to store metadata about the node lifecycle.
/// [`NodeWithMetadata`] automatically dereferences to `Node` via the [`Deref`] trait.
/// The node's status is set to [`NodeStatus::Dirty`] when a mutable reference is requested.
/// Accessing a deleted node will panic.
```

Wording taken from https://doc.rust-lang.org/std/sync/struct.Arc.html#deref-behavior.